tools/topology/topology1/development/sof-imx8-compr-pcm-wm8960.m4

```diff
@@ -0,0 +1,92 @@
+#
+# Topology with codec_adapter processing component for i.MX8QM/i.MX8QXP
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`sai.m4')
+include(`pcm.m4')
+include(`buffer.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include DSP configuration
+include(`platform/imx/imx8.m4')
+
+
+# Post process setup config
+
+ #codec Post Process setup config
+#
+# Define the pipelines
+#
+# PCM0 ----> volume -----> SAI3 (wm8960)
+#
+
+DECLARE_SOF_RT_UUID("passthrough_codec", passthrough_uuid, 0x376b5e44, 0x9c82,
+		0x4ec2, 0xbc, 0x83, 0x10, 0xea, 0x10, 0x1a, 0xf8, 0x8f);
+
+define(`CA_UUID', passthrough_uuid)
+
+# Post process setup config for playback
+define(`CA_SETUP_CONTROLBYTES',
+``	bytes "0x53,0x4f,0x46,0x00,0x00,0x00,0x00,0x00,'
+`	0x0C,0x00,0x00,0x00,0x00,0x10,0x00,0x03,'
+`	0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
+`	0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
+`	0x00,0x00,0x00,0x00,0x0C,0x00,0x00,0x00,'
+`	0x20,0x00,0x00,0x00"''
+)
+
+define(`CA_SETUP_CONTROLBYTES_MAX', 300)
+
+undefine(`DAI_PERIODS')
+define(`DAI_PERIODS', 8)
+
+dnl PIPELINE_PCM_ADD(pipeline,
+dnl     pipe id, pcm, max channels, format,
+dnl     period, priority, core,
+dnl     pcm_min_rate, pcm_max_rate, pipeline_rate,
+dnl     time_domain, sched_comp)
+
+# Low Latency playback pipeline 1 on PCM 0 using max 2 channels of s32le.
+# Set 1000us deadline with priority 0 on core 0
+PIPELINE_PCM_ADD(sof/pipe-codec-adapter-playback.m4,
+	1, 0, 2, s32le,
+	1000, 0, 0,
+	48000, 48000, 48000)
+
```

Comment:
The only difference between capture and playback topologies is 'playback' and 'capture' inside pipe-codec-adapter-playback.m4 file right?

So, better create a single file which is compiled into 2 different `tplg` with a parameter.

Eg.

```
PIPELINE_PCM_ADD(sof/pipe-codec-adapter-`DIR`.m4
```

You can take some inspiration from this commit: 52bc39fe73f96c072d41abef510b7c239a8d395c

Reply:
Could do that but would have to also add the following parameters:

1) BUFFER
```
DAI_ADD(sof/pipe-dai-`DIR`.m4,
	1, SAI, 1, sai1-wm8960-hifi,
	`BUFFER`, 2, s32le,
	1000, 0, 0, SCHEDULE_TIME_DOMAIN_DMA)
```

2) DIR_MACRO

```
COMPR_`DIR_MACRO`_ADD(Port0, 0, PIPELINE_PCM_1)
```

So that would be a total of 3 parameters. (assuming that what I wrote above would actually work) Would that still be preferable?

Reply:
@LaurentiuM1234 you are right, is not that easy.  Let's drop this idea.

Reply:
I mean this idea _could_ be doable if one parameter comes from outside and the others are defined based on it (using M4's ifelse statement). Up to you if you want to do it or not though.

Reply:
Wouldn't mind giving it a shot. Maybe even do something like (again, if this is actually possible):

```
if DIR == playback:
    PIPELINE_PCM_ADD(playback)
    DAI_ADD(playback)
    COMPR_PLAYBACK_ADD(...)
else:
    PIPELINE_PCM_ADD(capture)
    DAI_ADD(capture)
    COMPR_CAPTURE_ADD(...)
```

so as to use only 1 parameter. @dbaluta What's your take on this? Do you think it would be worth it?

Reply:
You'd have e.g. ``define(`DIRMACRO`, ifelse(DIR, playback, `PLAYBACK`, `CAPTURE`))``. My syntax might be wrong but it's something like that, and you'd do it at the beginning of the file.

Reply:
Yes, this would work. I'm not familiar enough with .m4 language to asses how hard would be to implement it. 

but if you can do it relatively quick then go for it.